frontend/app/src/components/Countdown/Countdown.tsx

```diff
@@ -14,58 +14,40 @@
  * limitations under the License.
  */
 
-import React, { PureComponent, ReactNode } from "react"
+import React, { useState } from "react"
 
 import { StyledCountdown } from "./styled-components"
 
 interface Props {
   countdown: number
-  endCallback: () => void
+  endCallback?: () => void
 }
 
-interface State {
-  countdown: number
-}
-
-class Countdown extends PureComponent<Props, State> {
-  public static defaultProps: Partial<Props> = {
-    endCallback: () => {},
-  }
+const Countdown: React.FC<Props> = ({ countdown, endCallback = () => {} }) => {
+  const [currentCountdown, setCurrentCountdown] = useState(countdown)
 
-  state = {
-    countdown: this.props.countdown,
-  }
-
-  onAnimationEnd = async (): Promise<any> => {
-    const { countdown } = this.state
-    const { endCallback } = this.props
-    const newCountdown = countdown - 1
+  const onAnimationEnd = (): void => {
```

Comment:
Do you know why `onAnimationEnd` was an `async` function before?

Reply:
Nope, not sure. But I changed it back to async.

Reply:
I have just had a look at the MDN web docs and it does not mention asynchronicity: https://developer.mozilla.org/en-US/docs/Web/API/Element/animationend_event
So in this case we should probably remove it to avoid confusion

Reply:
Oki, I removed it again 👍